### PR TITLE
fix(core): source installation script fixes

### PR DESCRIPTION
**Pull request by sc979**
## Description

1- Correct source installation script  (eg: ubuntu and debian) :
- randomize the APP_SECRET in .env files
- deploy the .env files to the right location
- correct centreon_cachedir in the conf file for web installation
- modify the rights on the /var/cache/centreon folder
- replace missing macros in some configuration files in the centreon's /bin folder

2- Update documentation

**Fixes** # (MON-4674 | https://github.com/centreon/centreon/issues/8249 , https://github.com/centreon/centreon/pull/8180 , https://github.com/centreon/centreon/issues/8252 )

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Check the issues and try to install the IHM using the sources on a Debian Buster OS

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: doc/en/installation/from_sources.rst [hunk 1]
```diff
@@ -697,7 +697,7 @@ Centreon SNMP trap management installation
     /etc/snmp                                                  OK
 
     Where is your CentreonTrapd binaries directory ?
-    default to [/usr/local/centreon/bin]
+    default to [/usr/share/centreon/bin]
     > /usr/share/centreon/bin
```
Comment (sc979) [inline doc/en/installation/from_sources.rst]:
```suggestion
    >
```

File: doc/fr/installation/from_sources.rst [hunk 2]
```diff
@@ -703,7 +703,7 @@ Installation du système de gestion des traps SNMP (CentreonTrapD)
     /etc/snmp                                                  OK
 
     Where is your CentreonTrapd binaries directory ?
-    default to [/usr/local/centreon/bin]
+    default to [/usr/share/centreon/bin]
     > /usr/share/centreon/bin
```
Comment (sc979) [inline doc/fr/installation/from_sources.rst]:
```suggestion
    >
```

File: install.sh [hunk 3]
```diff
@@ -414,6 +412,10 @@ fi
 purge_centreon_tmp_dir "silent"
 server=$(hostname -f)
 
+# Replace global variables
+
```
Comment (sc979) [inline install.sh]:
```suggestion
```

File: install.sh [hunk 4]
```diff
@@ -414,6 +412,10 @@ fi
...
+
```
Comment (sc979) [inline install.sh]:
```suggestion
```

File: install.sh [hunk 5]
```diff
@@ -414,6 +412,10 @@ fi
...
+
```
Comment (sc979) [inline install.sh]:
```suggestion
```

File: libinstall/CentWeb.sh [hunk 6]
```diff
@@ -301,7 +334,7 @@ ${CAT} "$file_perl_temp" | while read file ; do
 done
 check_result $flg_error "$(gettext "Change macros for perl binary")"
 
-### Step 3: Change right on monitoringengine_etcdir
+### Step 3: Change right on monitoringengine_/etc/centreon folder
```
Comment (sc979) [inline libinstall/CentWeb.sh]:
```suggestion
### Step 3: Change right on monitoringengine /etc/centreon folder
```

File: libinstall/CentWeb.sh [hunk 7]
```diff
@@ -317,7 +350,7 @@ find "$MONITORINGENGINE_ETC" -type f -print | \
 [ $? -ne 0 ] && flg_error=1
 check_result $flg_error "$(gettext "Change right on") $MONITORINGENGINE_ETC"
 
-### Change right to broker_etcdir
+### Change right to broker_/etc/centreon-broker folder
```
Comment (sc979) [inline libinstall/CentWeb.sh]:
```suggestion
### Change right to broker /etc/centreon-broker folder
```

File: libinstall/functions [hunk 8]
```diff
@@ -2367,10 +2401,10 @@ clean_sudo() {
 #----
 ## Find file with macro in directory.
 ## @param	macro (separate with comma)
-## @param 	directoty where to find
+## @param 	directory where to find
```
Comment (sc979) [inline libinstall/functions]:
```suggestion
## @param 	folder where to find
```

File: doc/en/installation/common/install_packages.rst [hunk 9]
```diff
@@ -135,14 +135,16 @@ option in **/etc/my.cnf** because it will *not* work. Run the commands:
...
+    # echo -ne "[Service]\nopen_files_limit=32000\n" | tee /etc/systemd/system/mariadb.service.d/centreon.conf
     # systemctl daemon-reload
     # systemctl restart mysql
 
 **For MySQL**: ::
 
     # mkdir -p  /etc/systemd/system/mysqld.service.d
-    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mysqld.service.d/limits.conf
+    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mysqld.service.d/centreon.conf
+    # echo -ne "[Service]\nopen_files_limit=32000\n" | tee /etc/systemd/system/mysqld.service.d/centreon.conf
```
Comment (sc979) [inline doc/en/installation/common/install_packages.rst]:
```suggestion
```

File: doc/fr/installation/common/install_packages.rst [hunk 10]
```diff
@@ -140,13 +140,15 @@ cette option dans /etc/my.cnf *ne fonctionnera pas*.
 
     # mkdir -p  /etc/systemd/system/mariadb.service.d/
     # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mariadb.service.d/limits.conf
+    # echo -ne "[Service]\nopen_files_limit=32000\n" | tee /etc/systemd/system/mariadb.service.d/centreon.conf
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
```

File: doc/fr/installation/common/install_packages.rst [hunk 11]
```diff
@@ -140,13 +140,15 @@ cette option dans /etc/my.cnf *ne fonctionnera pas*.
 
     # mkdir -p  /etc/systemd/system/mariadb.service.d/
     # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mariadb.service.d/limits.conf
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mariadb.service.d/centreon.conf
```

File: doc/fr/installation/common/install_packages.rst [hunk 12]
```diff
@@ -140,13 +140,15 @@ cette option dans /etc/my.cnf *ne fonctionnera pas*.
...
+    # echo -ne "[Service]\nopen_files_limit=32000\n" | tee /etc/systemd/system/mariadb.service.d/centreon.conf
     # systemctl daemon-reload
     # systemctl restart mysql
 
 **Pour MySQL** : ::
 
     # mkdir -p  /etc/systemd/system/mysqld.service.d
     # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mysqld.service.d/limits.conf
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mysqld.service.d/centreon.conf
```

File: doc/fr/installation/common/install_packages.rst [hunk 13]
```diff
@@ -140,13 +140,15 @@ cette option dans /etc/my.cnf *ne fonctionnera pas*.
...
+    # echo -ne "[Service]\nopen_files_limit=32000\n" | tee /etc/systemd/system/mysqld.service.d/centreon.conf
```
Comment (sc979) [inline doc/fr/installation/common/install_packages.rst]:
```suggestion
```

File: doc/en/installation/common/install_packages.rst [hunk 14]
```diff
@@ -135,14 +135,16 @@ option in **/etc/my.cnf** because it will *not* work. Run the commands:
 **For MariaDB**: ::
 
     # mkdir -p  /etc/systemd/system/mariadb.service.d/
-    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mariadb.service.d/limits.conf
+    # echo -ne "[Service]\nLimitNOFILE=32000\n" | tee /etc/systemd/system/mariadb.service.d/centreon.conf
+    # echo -ne "[Service]\nopen_files_limit=32000\n" | tee /etc/systemd/system/mariadb.service.d/centreon.conf
```
Comment (sc979) [inline doc/en/installation/common/install_packages.rst]:
```suggestion
```

File: libinstall/CentWeb.sh [hunk 15]
```diff
@@ -245,20 +274,22 @@ ${CAT} "$file_php_temp" | while read file ; do
...
         [ $? -ne 0 ] && flg_error=1
     log "MACRO" "$(gettext "Copy in final dir") : $file"
     cp -f $TMP_DIR/work/$file $TMP_DIR/final/$file >> $LOG_FILE 2>&1
 done
 check_result $flg_error "$(gettext "Change macros for php files")"
 
-macros="@CENTREON_ETC@,@CENTREON_CACHEDIR@,@CENTPLUGINSTRAPS_BINDIR@,@CENTREON_LOG@,@CENTREON_VARLIB@,@CENTREONTRAPD_BINDIR@"
+macros="@CENTREON_ETC@,@CENTREON_CACHEDIR@,@CENTPLUGINSTRAPS_BINDIR@,@CENTREON_LOG@,@CENTREON_VARLIB@,@CENTREONTRAPD_BINDIR@,%APP_SECRET%"
 find_macros_in_dir "$macros" "$TMP_DIR/src" "config" "*.php*" "file_php_config_temp"
+find_macros_in_dir "$macros" "$TMP_DIR/src/" "." ".env*" "file_env_temp"
```
Comment (sc979) [inline libinstall/CentWeb.sh]:
```suggestion
```

File: libinstall/CentWeb.sh [hunk 16]
```diff
@@ -388,9 +420,12 @@ check_result $? "$(gettext "Change right for install directory")"
 cp -f $TMP_DIR/final/bootstrap.php $INSTALL_DIR_CENTREON/bootstrap.php >> "$LOG_FILE" 2>&1
 $CHOWN $WEB_USER:$WEB_GROUP $INSTALL_DIR_CENTREON/bootstrap.php
 
-cp -f $TMP_DIR/final/.env $INSTALL_DIR_CENTREON/.env >> "$LOG_FILE" 2>&1
+cp -f $TMP_DIR/final/.env $INSTALL_DIR_CENTREON >> "$LOG_FILE" 2>&1
```
Comment (sc979) [inline libinstall/CentWeb.sh]:
```suggestion
cp -f $TMP_DIR/final/.env $INSTALL_DIR_CENTREON/.env >> "$LOG_FILE" 2>&1
```

File: libinstall/CentWeb.sh [hunk 17]
```diff
@@ -388,9 +420,12 @@ check_result $? "$(gettext "Change right for install directory")"
...
 $CHOWN $WEB_USER:$WEB_GROUP $INSTALL_DIR_CENTREON/.env
 
+cp -f $TMP_DIR/final/.env.local.php $INSTALL_DIR_CENTREON >> "$LOG_FILE" 2>&1
```
Comment (sc979) [inline libinstall/CentWeb.sh]:
```suggestion
cp -f $TMP_DIR/final/.env.local.php $INSTALL_DIR_CENTREON/.env.local.php >> "$LOG_FILE" 2>&1
```

File: libinstall/CentWeb.sh [hunk 18]
```diff
@@ -587,11 +621,6 @@ $INSTALL_DIR/cinstall $cinstall_opts \
 check_result $? "$(gettext "Install centreonSyncArchives")"
 
 ## Install generateSqlLite
-log "INFO" "$(gettext "Prepare generateSqlLite")"
-cp $TMP_DIR/src/bin/generateSqlLite \
-    $TMP_DIR/final/bin/generateSqlLite >> "$LOG_FILE" 2>&1
```
Comment (cgagnaire) [inline libinstall/CentWeb.sh]: Are you sure the files are copied into final directory ? (Needed just after)

File: libinstall/CentWeb.sh [hunk 19]
```diff
@@ -226,16 +251,20 @@ ${CAT} "$file_sql_temp" | while read file ; do
...
 find_macros_in_dir "$macros" "$TMP_DIR/src/" "www" "*.php" "file_php_temp"
-
-log "INFO" "$(gettext "Apply macros")"
+find_macros_in_dir "$macros" "$TMP_DIR/src/" "bin" "*" "file_bin_temp"
+log "INFO" "$(gettext "Apply macros on php files")"
 
 flg_error=0
-${CAT} "$file_php_temp" | while read file ; do
+${CAT} "$file_php_temp" "$file_bin_temp" | while read file ; do
     log "MACRO" "$(gettext "Change macro for") : $file"
```
Comment (cgagnaire) [inline libinstall/CentWeb.sh]: log "MACRO" seems odd.

File: libinstall/CentWeb.sh [hunk 20]
```diff
@@ -245,20 +274,21 @@ ${CAT} "$file_php_temp" | while read file ; do
...
 find_macros_in_dir "$macros" "$TMP_DIR/src" "config" "*.php*" "file_php_config_temp"
-
-log "INFO" "$(gettext "Apply macros")"
+find_macros_in_dir "$macros" "$TMP_DIR/src/" "." ".env*" "file_env_temp"
+log "INFO" "$(gettext "Apply macros on env and config files")"
 
 flg_error=0
-${CAT} "$file_php_config_temp" | while read file ; do
+${CAT} "$file_php_config_temp" "$file_env_temp" | while read file ; do
         log "MACRO" "$(gettext "Change macro for") : $file"
```
Comment (cgagnaire) [inline libinstall/CentWeb.sh]: log "MACRO" seems odd.

File: doc/en/installation/from_sources.rst [hunk 21]
```diff
@@ -508,23 +500,28 @@ PHP FPM configuration
     Do you want to reload PHP FPM service ?
     [y/n], default to [n]:
     > y
-    Reloading PHP FPM service                                  OK
+
+    Do you want to reload PHP FPM service ?
```
Comment (cgagnaire) [inline doc/en/installation/from_sources.rst]: Same question duplicated.

File: libinstall/CentWeb.sh (hunk 18 above)
Comment (sc979) [reply libinstall/CentWeb.sh]: yes they are.
from the temporary src (original file) to the final

File: libinstall/CentWeb.sh (hunk 19 above)
Comment (sc979) [reply libinstall/CentWeb.sh]: it's only a param to specify in the logs what it is about.
The installation log can be found in the centreon's source folder,
eg : centreon-web-19.10.6/log/install_centreon.log
Here how it's processed : https://github.com/centreon/centreon/blob/master/libinstall/functions#L121
Do you want this param value to be changed to something more consistent ?
image

File: doc/en/installation/from_sources.rst (hunk 21 above)
Comment (sc979) [reply doc/en/installation/from_sources.rst]: nice catch ;)

File: libinstall/CentWeb.sh (hunk 19 above)
Comment (cgagnaire) [reply libinstall/CentWeb.sh]: I think it's meant to be INFO.

**Pull request merged**